geomstats/geometry/general_linear.py

```diff
@@ -1,192 +1,125 @@
 
-        Group exponential of the Lie group of all invertible matrices.
+        The orbit is returned as the path function satisfying:
 
         Parameters
         ----------
-        tangent_vec
-        base_point
-        point_type
+        point : array-like, shape=[n, n]
+            Target point.
+        base_point : array-like, shape=[n, n], optional
+            Base point. Defaults to identity.
 
         Returns
         -------
-        group_exp
-        """
-        tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=3)
-        base_point = gs.to_ndarray(base_point, to_ndim=3)
+        path : callable
+            The one-parameter orbit.
+            Satisfies `path(0) = base_point` and `point(1) = point`.
 
-        tangent_vec_at_identity = self.compose(
-            self.inverse(base_point), tangent_vec)
+        Note
+        ----
+        Denoting `point` by :math: `g` and `base_point` by :math: `h`,
+        the orbit :math: `\gamma` satisfies:
 
-        exp_from_identity = self.exp_from_identity(
-                tangent_vec_at_identity)
+        .. math::
 
-        group_exp = self.compose(
-            base_point, exp_from_identity)
+            \gamma(t) = {\mathrm e}^{t X} \cdot h \\
+            \quad {\mathrm with} \quad\\
+            {\mathrm e}^{X} = g h^{-1}
 
-        return group_exp
+        The path is not uniquely defined and depends on
+        the choice of :math: `V` returned by `class.log`.
 
-    def log_from_identity(self, point, point_type=None):
-        """Compute group logarithm at the identity.
+        Vectorization
+        -------------
+        Return a collection of trajectories (4-D array)
+        from a collection of input matrices (3-D array).
 
-        Group logarithm of the Lie group of
-        all invertible matrices at the identity.
-
-        Parameters
-        ----------
-        point
-        point_type
-
-        Returns
-        -------
-        group_log
+        Will work when expm gets properly 4-D vectorized.
         """
-        point = gs.to_ndarray(point, to_ndim=3)
-        group_log = gs.linalg.logm(point)
-
-        return gs.real(group_log)
-
-    def log_not_from_identity(
-            self, point, base_point, point_type=None):
-        """Compute group logarithm at a base point.
-
-        Group logarithm of the Lie group of all invertible matrices.
-
-        Parameters
-        ----------
-        point
-        base_point
-        point_type
-
-        Returns
-        -------
-        group_log
-        """
-        point = gs.to_ndarray(point, to_ndim=3)
-        base_point = gs.to_ndarray(base_point, to_ndim=3)
-
-        point_near_identity = self.compose(
-            self.inverse(base_point), point)
-
-        log_from_identity = self.log_from_identity(
-            point_near_identity)
+        vector = cls.log(point, base_point)
 
-        group_log = self.compose(
-            base_point, log_from_identity)
+        def path(time):
```

Comment:
maybe `point_on_orbit` to micmic the geodesic's API? https://github.com/geomstats/geomstats/blob/master/geomstats/geometry/riemannian_metric.py#L261

Reply:
(or change geodesic's API to `path`, whatever is more understandable)

Reply:
Hm yes I've seen this, a bit obscure don't you think? 
I would expect `point_on_orbit` to test whether point is in the orbit

Reply:
just renamed this everywhere :)
btw doing so I noticed that `SPD.geodesic` does not behave as expected and has prototype:
```
SPD.geodesic(point, vector) => path
```
I'll put this in separate issues as you suggested, just to mention. Do we agree that this should be either:
+ `geodesic(point2, base_point) -> path`
+ `geodesic(vector, base_point) -> path` ?

Reply:
Clearly not the time,  but we might want to think about having two names, like `interpolate` vs `shoot` someday...
If you have any idea ^^